### PR TITLE
Added `EatingScp330EventArgs` and `EatenScp330EventArgs`

### PR DESCRIPTION
**Pull request by Marco15453**
# **REQUIRES MORE TESTING**

# Events
- Added EatingScp330EventArgs (**If cancelled, the Animation will not stop playing**)
- Added EatenScp330EventArgs

### CODE REVIEW

**Review by joker-119 — CHANGES_REQUESTED** [review]
Please use proper English grammar and spelling in documentation.

**Comment by TheUltiOne** [comment]
I think it should be EatingScp330EventArgs to fit with how SCP is written everywhere in EXILED, but that’s just an opinion.

**Review by joker-119 — APPROVED** [review]

**Review by joker-119 — CHANGES_REQUESTED** [review]
Fix build errors.
Stop editing PR's through github.
Also change the file names to patch the class names.

**Review by joker-119 — CHANGES_REQUESTED** [review]
Please at least attempt to build the solution yourself in release, there are still build errors.

File: Exiled.Events/Patches/Events/Scp330/Eaten.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    using NorthwoodLib.Pools;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="Scp330Bag.ServerOnUsingCompleted"/>.
+    /// Adds the <see cref="Handlers.Scp330.OnEaten"/> event.
+    /// </summary>
+    [HarmonyDebug]
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eaten.cs:30]: Remove this shit

File: Exiled.Events/Patches/Events/Scp330/Eaten.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                new CodeInstruction(OpCodes.Ldloc_0),
+
+                // var ev = new EatenSCP330EventArgs(player, candy)
+                new CodeInstruction(OpCodes.Newobj, GetDeclaredConstructors(typeof(EatenScp330EventArgs))[0]),
+
+                // Handlers.SCP330.OnEaten(ev)
+                new CodeInstruction(OpCodes.Call, Method(typeof(Handlers.Scp330), nameof(Handlers.Scp330.OnEaten))),
+            });
+
+            for (int i = 0; i < newInstructions.Count; i++)
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eaten.cs:58]: Use `z` instead

File: Exiled.Events/Patches/Events/Scp330/Eating.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
+// -----------------------------------------------------------------------
+// <copyright file="Eating.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Events.Scp330
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eating.cs:8]: Do not make multiple files for the same method, just make a transpiler that implements both events.

File: Exiled.Events/Patches/Events/Scp330/Eating.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    using InventorySystem.Items.Usables.Scp330;
+
+    using NorthwoodLib.Pools;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="Scp330Bag.ServerOnUsingCompleted"/>.
+    /// Adds the <see cref="Handlers.Scp330.OnEating"/> event.
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eating.cs:28]: `Handlers.Scp330.Eating` is the event

File: Exiled.Events/Patches/Events/Scp330/Eating.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+                // if(!ev.IsAllowed)
+                //  return
+                new CodeInstruction(OpCodes.Callvirt, PropertyGetter(typeof(EatingScp330EventArgs), nameof(EatingScp330EventArgs.IsAllowed))),
+                new CodeInstruction(OpCodes.Brfalse, returnLabel),
+            });
+
+            newInstructions[newInstructions.Count - 1].WithLabels(returnLabel);
+
+            for (int i = 0; i < newInstructions.Count; i++)
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eating.cs:71]: Use `z` instead.

File: Exiled.Events/Patches/Events/Scp330/Eating.cs [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+            const int offset = -3;
+
+            var index = newInstructions.FindIndex(instruction => instruction.opcode == OpCodes.Callvirt && (MethodInfo)instruction.operand == Method(typeof(ICandy), nameof(ICandy.ApplyEffects))) + offset;
+
+            var returnLabel = generator.DefineLabel();
+
+            newInstructions.InsertRange(index, new[]
+            {
+                // Player.Get
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eating.cs:45]: If you want to comment your own instructions, comment everything.
IE `Player.Get(args)`

File: Exiled.Events/Patches/Events/Scp330/Eaten.cs [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+                new CodeInstruction(OpCodes.Callvirt, PropertyGetter(typeof(Scp330Bag), nameof(Scp330Bag.Owner))),
+                new CodeInstruction(OpCodes.Call, Method(typeof(Player), nameof(Player.Get), new[] { typeof(ReferenceHub) })),
+
+                // ICandy
+                new CodeInstruction(OpCodes.Ldloc_0),
+
+                // var ev = new EatenSCP330EventArgs(player, candy)
+                new CodeInstruction(OpCodes.Newobj, GetDeclaredConstructors(typeof(EatenScp330EventArgs))[0]),
+
+                // Handlers.SCP330.OnEaten(ev)
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eaten.cs:54]: Match files' name, change from `SCP330` to `Scp330`

File: Exiled.Events/Patches/Events/Scp330/Eaten.cs [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+    using InventorySystem.Items.Usables.Scp330;
+
+    using NorthwoodLib.Pools;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="Scp330Bag.ServerOnUsingCompleted"/>.
+    /// Adds the <see cref="Handlers.Scp330.OnEaten"/> event.
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eaten.cs:28]: `Eaten` is the event

File: Exiled.Events/Handlers/Scp330.cs [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+        /// </summary>
+        public static event CustomEventHandler<EatingScp330EventArgs> Eating;
+
+        /// <summary>
+        /// Invoked after player ate SCP330.
+        /// </summary>
+        public static event CustomEventHandler<EatenScp330EventArgs> Eaten;
+
+        /// <summary>
+        /// Called before player ate SCP330.
```
Comment (NaoUnderscore) [inline Exiled.Events/Handlers/Scp330.cs:31]: I'd change to _Called before a player eats Scp330_

File: Exiled.Events/Handlers/Scp330.cs [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    /// </summary>
+    public static class Scp330
+    {
+        /// <summary>
+        /// Invoked before player ate SCP330.
+        /// </summary>
+        public static event CustomEventHandler<EatingScp330EventArgs> Eating;
+
+        /// <summary>
+        /// Invoked after player ate SCP330.
```
Comment (NaoUnderscore) [inline Exiled.Events/Handlers/Scp330.cs:26]: I'd change to _Invoked after the player has eaten Scp330_

File: Exiled.Events/Handlers/Scp330.cs [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+    using static Exiled.Events.Events;
+
+    /// <summary>
+    /// SCP330 related events.
+    /// </summary>
+    public static class Scp330
+    {
+        /// <summary>
+        /// Invoked before player ate SCP330.
```
Comment (NaoUnderscore) [inline Exiled.Events/Handlers/Scp330.cs:21]: I'd change to _Invoked before a player eats Scp330_

File: Exiled.Events/Handlers/Scp330.cs [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace Exiled.Events.Handlers
+{
+    using Exiled.Events.EventArgs;
+    using Exiled.Events.Extensions;
+
+    using static Exiled.Events.Events;
+
+    /// <summary>
+    /// SCP330 related events.
```
Comment (NaoUnderscore) [inline Exiled.Events/Handlers/Scp330.cs:16]: Match naming rules for Scp330

File: Exiled.Events/Handlers/Scp330.cs [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+        public static event CustomEventHandler<EatenScp330EventArgs> Eaten;
+
+        /// <summary>
+        /// Called before player ate SCP330.
+        /// </summary>
+        /// <param name="ev">The <see cref="EatingScp330EventArgs"/> instance.</param>
+        public static void OnEating(EatingScp330EventArgs ev) => Eating.InvokeSafely(ev);
+
+        /// <summary>
+        /// Called after player ate SCP330.
```
Comment (NaoUnderscore) [inline Exiled.Events/Handlers/Scp330.cs:37]: I'd change to Called after the player has eaten Scp330

File: Exiled.Events/Handlers/Scp330.cs [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+        /// </summary>
+        /// <param name="ev">The <see cref="EatenScp330EventArgs"/> instance.</param>
+        public static void OnEaten(EatenScp330EventArgs ev) => Eaten.InvokeSafely(ev);
```
Comment (NaoUnderscore) [inline Exiled.Events/Handlers/Scp330.cs:40]: I'd change to `OnEatenScp330`

File: Exiled.Events/Handlers/Scp330.cs [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+        /// <summary>
+        /// Invoked after player ate SCP330.
+        /// </summary>
+        public static event CustomEventHandler<EatenScp330EventArgs> Eaten;
+
+        /// <summary>
+        /// Called before player ate SCP330.
+        /// </summary>
+        /// <param name="ev">The <see cref="EatingScp330EventArgs"/> instance.</param>
+        public static void OnEating(EatingScp330EventArgs ev) => Eating.InvokeSafely(ev);
```
Comment (NaoUnderscore) [inline Exiled.Events/Handlers/Scp330.cs:34]: I'd change to `OnEatingScp330`

File: Exiled.Events/Patches/Events/Scp330/Eaten.cs [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            const int offset = -1;
+            var index = newInstructions.FindIndex(instruction => instruction.opcode == OpCodes.Call && (MethodInfo)instruction.operand == Method(typeof(Scp330Bag), nameof(Scp330Bag.ServerRefreshBag))) + offset;
+
+            newInstructions.InsertRange(index, new[]
+            {
+                // Player.Get
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eaten.cs:43]: Extend this comment with args used for `Player::Get`

File: Exiled.Events/Patches/Events/Scp330/Eating.cs [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+                new CodeInstruction(OpCodes.Ldarg_0).MoveLabelsFrom(newInstructions[index]),
+                new CodeInstruction(OpCodes.Callvirt, PropertyGetter(typeof(Scp330Bag), nameof(Scp330Bag.Owner))),
+                new CodeInstruction(OpCodes.Call, Method(typeof(Player), nameof(Player.Get), new[] { typeof(ReferenceHub) })),
+
+                // ICandy
+                new CodeInstruction(OpCodes.Ldloc_0),
+
+                // True
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eating.cs:53]: Match C# naming rules, this is not python.
`True` -> `true`

File: Exiled.Events/Patches/Events/Scp330/Eating.cs [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+                new CodeInstruction(OpCodes.Ldc_I4_1),
+
+                // var ev = new EatingSCP330EventArgs(player, candy, true)
+                new CodeInstruction(OpCodes.Newobj, GetDeclaredConstructors(typeof(EatingScp330EventArgs))[0]),
+                new CodeInstruction(OpCodes.Dup),
+
+                // Handlers.SCP330.OnEating(ev)
+                new CodeInstruction(OpCodes.Call, Method(typeof(Handlers.Scp330), nameof(Handlers.Scp330.OnEating))),
+
+                // if(!ev.IsAllowed)
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eating.cs:63]: Insert a space before the condition

File: Exiled.Events/EventArgs/EatingSCP330EventArgs.cs [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+namespace Exiled.Events.EventArgs
+{
+    using System;
+
+    using Exiled.API.Features;
+
+    using InventorySystem.Items.Usables.Scp330;
+
+    /// <summary>
+    /// Contains all informations before a player eats SCP330.
```
Comment (NaoUnderscore) [inline Exiled.Events/EventArgs/EatingSCP330EventArgs.cs:17]: Match naming rules for `Scp330`

File: Exiled.Events/EventArgs/EatingSCP330EventArgs.cs [hunk 20]
```diff
@@ -0,0 +1,49 @@
+// -----------------------------------------------------------------------
+// <copyright file="EatingScp330EventArgs.cs" company="Exiled Team">
```
Comment (NaoUnderscore) [inline Exiled.Events/EventArgs/EatingSCP330EventArgs.cs:2]: Fix build errors by changing the file name.
Header should match file name.

File: Exiled.Events/EventArgs/EatenSCP330EventArgs.cs [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+    /// <summary>
+    /// Contains all informations after a player eaten SCP330.
+    /// </summary>
+    public class EatenScp330EventArgs : EventArgs
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EatenScp330EventArgs"/> class.
+        /// </summary>
+        /// <param name="player"><inheritdoc cref="Player"/>.</param>
+        /// <param name="candy"><inheritdoc cref="ICandy"/>.</param>
```
Comment (NaoUnderscore) [inline Exiled.Events/EventArgs/EatenSCP330EventArgs.cs:25]: `<inheritdoc cref="Candy"/>`

File: Exiled.Events/EventArgs/EatenSCP330EventArgs.cs [hunk 22]
```diff
@@ -0,0 +1,42 @@
...
+namespace Exiled.Events.EventArgs
+{
+    using System;
+
+    using Exiled.API.Features;
+
+    using InventorySystem.Items.Usables.Scp330;
+
+    /// <summary>
+    /// Contains all informations after a player eaten SCP330.
```
Comment (NaoUnderscore) [inline Exiled.Events/EventArgs/EatenSCP330EventArgs.cs:17]: I'd change to _Contains all informations after a player has eaten Scp330._

File: Exiled.Events/EventArgs/EatingSCP330EventArgs.cs [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+        /// <param name="isAllowed"><see cref="IsAllowed"/>.</param>
+        public EatingScp330EventArgs(Player player, ICandy candy, bool isAllowed = true)
+        {
+            Player = player;
+            Candy = candy;
+            IsAllowed = isAllowed;
+        }
+
+        /// <summary>
+        /// Gets the player who's eating SCP330.
```
Comment (NaoUnderscore) [inline Exiled.Events/EventArgs/EatingSCP330EventArgs.cs:35]: Match naming rules for `Scp330`

File: Exiled.Events/EventArgs/EatingSCP330EventArgs.cs [hunk 24]
```diff
@@ -0,0 +1,49 @@
...
+        /// </summary>
+        public Player Player { get; }
+
+        /// <summary>
+        /// Gets the <see cref="ICandy"/> that the player is eating.
```
Comment (NaoUnderscore) [inline Exiled.Events/EventArgs/EatingSCP330EventArgs.cs:40]: I'd change to _Get the candy that is being eaten by the player_

File: Exiled.Events/EventArgs/EatingSCP330EventArgs.cs [hunk 25]
```diff
@@ -0,0 +1,49 @@
...
+        /// </summary>
+        public ICandy Candy { get; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether or not player is allowed to eat SCP330.
```
Comment (NaoUnderscore) [inline Exiled.Events/EventArgs/EatingSCP330EventArgs.cs:45]: I'd change to `Gets or sets a value indicating whether or not the player can eat Scp330.`

File: Exiled.Events/EventArgs/EatenSCP330EventArgs.cs [hunk 26]
```diff
@@ -0,0 +1,42 @@
...
+            Candy = candy;
+        }
+
+        /// <summary>
+        /// Gets the player who's eaten SCP330.
+        /// </summary>
+        public Player Player { get; }
+
+        /// <summary>
+        /// Gets the <see cref="ICandy"/> that the player has eaten.
```
Comment (NaoUnderscore) [inline Exiled.Events/EventArgs/EatenSCP330EventArgs.cs:38]: I'd change to `Get the <see cref="ICandy"/> that was eaten by the player`

File: Exiled.Events/EventArgs/EatenSCP330EventArgs.cs [hunk 27]
```diff
@@ -0,0 +1,42 @@
...
+        /// <param name="player"><inheritdoc cref="Player"/>.</param>
+        /// <param name="candy"><inheritdoc cref="ICandy"/>.</param>
+        public EatenScp330EventArgs(Player player, ICandy candy)
+        {
+            Player = player;
+            Candy = candy;
+        }
+
+        /// <summary>
+        /// Gets the player who's eaten SCP330.
```
Comment (NaoUnderscore) [inline Exiled.Events/EventArgs/EatenSCP330EventArgs.cs:33]: Match naming rules for `Scp330`

Comment (NaoUnderscore) [reply Exiled.Events/EventArgs/EatenSCP330EventArgs.cs:33]: Also, _Get the player who ate Scp330_ was fine.

File: Exiled.Events/EventArgs/EatenSCP330EventArgs.cs [hunk 28]
```diff
@@ -0,0 +1,42 @@
+// -----------------------------------------------------------------------
+// <copyright file="EatenScp330EventArgs.cs" company="Exiled Team">
```
Comment (NaoUnderscore) [inline Exiled.Events/EventArgs/EatenSCP330EventArgs.cs:2]: Fix build errors by changing the file name.
Header should match file name.

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

File: Exiled.Events/Handlers/Scp330.cs [hunk 29]
```diff
@@ -0,0 +1,42 @@
...
+
+    /// <summary>
+    /// SCP330 related events.
+    /// </summary>
+    public static class Scp330
+    {
+        /// <summary>
+        /// Invoked before player ate SCP330.
+        /// </summary>
+        public static event CustomEventHandler<EatingScp330EventArgs> Eating;
```
Comment (NaoUnderscore) [inline Exiled.Events/Handlers/Scp330.cs:23]: I'd change to `EatingScp330`

File: Exiled.Events/Handlers/Scp330.cs [hunk 30]
```diff
@@ -0,0 +1,42 @@
...
+
+        /// <summary>
+        /// Invoked after player ate SCP330.
+        /// </summary>
+        public static event CustomEventHandler<EatenScp330EventArgs> Eaten;
```
Comment (NaoUnderscore) [inline Exiled.Events/Handlers/Scp330.cs:28]: I'd change to `EatenScp330`

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

File: Exiled.Events/Handlers/Scp330.cs (hunk 29 above)
Comment (joker-119) [reply Exiled.Events/Handlers/Scp330.cs:23]: Not to mention fixing the past-tense version of "eat" in future-tense event summary.

Comment (NaoUnderscore) [reply Exiled.Events/Handlers/Scp330.cs:23]: I mentioned everything Joker. Don't worry.

Comment (Marco15453) [reply Exiled.Events/Handlers/Scp330.cs:23]: I'm german sorry.

**Comment by Marco15453** [comment]
It builds! [Screenshot](https://share.4everyone.xyz/screenshot/e267b27837f5d7076beb3681934692a7.png)

**Comment by BuildBoy12** [comment]
Stylecop go brrrr

**Comment by joker-119** [comment]
> It builds! [Screenshot](https://share.4everyone.xyz/screenshot/e267b27837f5d7076beb3681934692a7.png)

D:\a\EXILED\EXILED\Exiled.Events\EventArgs\EatingSCP330EventArgs.cs(2,4): error SA1638: File header file name documentation should match file name.   you sure about that, kid?

**Comment by Marco15453** [comment]
Fucking GitHub Desktop, It didn't renamed my classes!

File: Exiled.Events/Patches/Events/Scp330/Eaten.cs [hunk 31]
```diff
@@ -0,0 +1,63 @@
...
+                new CodeInstruction(OpCodes.Callvirt, PropertyGetter(typeof(Scp330Bag), nameof(Scp330Bag.Owner))),
+                new CodeInstruction(OpCodes.Call, Method(typeof(Player), nameof(Player.Get), new[] { typeof(ReferenceHub) })),
+
+                // ICandy
+                new CodeInstruction(OpCodes.Ldloc_0),
+
+                // var ev = new EatenScp330EventArgs(player, candy)
+                new CodeInstruction(OpCodes.Newobj, GetDeclaredConstructors(typeof(EatenScp330EventArgs))[0]),
+
+                // Handlers.Scp330.OnEaten(ev)
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eaten.cs:53]: I'd change to `OnEatenScp330`

File: Exiled.Events/Patches/Events/Scp330/Eaten.cs [hunk 32]
```diff
@@ -0,0 +1,63 @@
...
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            const int offset = -1;
+            var index = newInstructions.FindIndex(instruction => instruction.opcode == OpCodes.Call && (MethodInfo)instruction.operand == Method(typeof(Scp330Bag), nameof(Scp330Bag.ServerRefreshBag))) + offset;
+
+            newInstructions.InsertRange(index, new[]
+            {
+                // Player.Get(args)
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eaten.cs:42]: `Player.Get(this.Owner)`

File: Exiled.Events/Patches/Events/Scp330/Eating.cs [hunk 33]
```diff
@@ -0,0 +1,77 @@
...
+    using NorthwoodLib.Pools;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="Scp330Bag.ServerOnUsingCompleted"/>.
+    /// Adds the <see cref="Handlers.Scp330.EatingScp330"/> event.
+    /// </summary>
+    [HarmonyPatch(typeof(Scp330Bag), nameof(Scp330Bag.ServerOnUsingCompleted))]
+    internal static class Eating
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eating.cs:31]: I'd change file name to `EatingScp330`

File: Exiled.Events/Patches/Events/Scp330/Eating.cs [hunk 34]
```diff
@@ -0,0 +1,77 @@
...
+
+            const int offset = -3;
+
+            var index = newInstructions.FindIndex(instruction => instruction.opcode == OpCodes.Callvirt && (MethodInfo)instruction.operand == Method(typeof(ICandy), nameof(ICandy.ApplyEffects))) + offset;
+
+            var returnLabel = generator.DefineLabel();
+
+            newInstructions.InsertRange(index, new[]
+            {
+                // Player.Get(args)
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eating.cs:45]: `Player.Get(this.Owner)`

File: Exiled.Events/Patches/Events/Scp330/Eating.cs [hunk 35]
```diff
@@ -0,0 +1,77 @@
...
+
+                // var ev = new EatingScp330EventArgs(player, candy, True)
+                new CodeInstruction(OpCodes.Newobj, GetDeclaredConstructors(typeof(EatingScp330EventArgs))[0]),
+                new CodeInstruction(OpCodes.Dup),
+
+                // Handlers.Scp330.OnEating(ev)
+                new CodeInstruction(OpCodes.Call, Method(typeof(Handlers.Scp330), nameof(Handlers.Scp330.OnEatingScp330))),
+
+                // if (!ev.IsAllowed)
+                //  return
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eating.cs:64]: `return;`

File: Exiled.Events/Patches/Events/Scp330/Eating.cs [hunk 36]
```diff
@@ -0,0 +1,77 @@
...
+                new CodeInstruction(OpCodes.Ldloc_0),
+
+                // True
+                new CodeInstruction(OpCodes.Ldc_I4_1),
+
+                // var ev = new EatingScp330EventArgs(player, candy, True)
+                new CodeInstruction(OpCodes.Newobj, GetDeclaredConstructors(typeof(EatingScp330EventArgs))[0]),
+                new CodeInstruction(OpCodes.Dup),
+
+                // Handlers.Scp330.OnEating(ev)
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eating.cs:60]: `OnEatingScp330`

File: Exiled.Events/Patches/Events/Scp330/Eating.cs [hunk 37]
```diff
@@ -0,0 +1,77 @@
...
+                new CodeInstruction(OpCodes.Callvirt, PropertyGetter(typeof(Scp330Bag), nameof(Scp330Bag.Owner))),
+                new CodeInstruction(OpCodes.Call, Method(typeof(Player), nameof(Player.Get), new[] { typeof(ReferenceHub) })),
+
+                // ICandy
+                new CodeInstruction(OpCodes.Ldloc_0),
+
+                // True
+                new CodeInstruction(OpCodes.Ldc_I4_1),
+
+                // var ev = new EatingScp330EventArgs(player, candy, True)
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/Eating.cs:56]: `True` -> `true`

File: Exiled.Events/Patches/Events/Scp330/Eating.cs (hunk 17 above)
Comment (NaoUnderscore) [reply Exiled.Events/Patches/Events/Scp330/Eating.cs:53]: Match C# naming rules, this is not python.
`True` -> `true`

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

**Comment by TheUltiOne** [comment]
Wouldn’t it be Scp330Eaten or AteScp330???

**Comment by joker-119** [comment]
> Wouldn’t it be Scp330Eaten or AteScp330???

It could be either of the 3, technically "AteScp330" is the most gramatically correct, but I'm willing to overlook that to have consistency with naming.

**Review by joker-119 — APPROVED** [review]

**Review by BuildBoy12 — APPROVED** [review]

**Pull request merged**